core/shared/src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -743,8 +743,12 @@ sealed trait ZPure[+W, -S1, +S2, -R, +E, +A] { self =>
           zPure.flag match {
             case FlagType.ClearLogOnError =>
               clearLogOnError.push(zPure.value)
-              curZPure = zPure.continue.fold(
-                e => { clearLogOnError.popOrElse(false); e },
+              curZPure = zPure.continue.bimap(
```

Comment:
Using `fold` was wrong as it was recovering from the effect, used `bimap` instead.